### PR TITLE
CB-10808 CLI Should handle templates with files located in sub directory

### PR DESCRIPTION
**Pull request by csantanapr**
Refer to [CB-10808](https://issues.apache.org/jira/browse/CB-10808) for more info

### CODE REVIEW

**Comment by csantanapr** [comment]
@stevengill @vladimir-kotikov  could you review this when you have chance.

File: cordova-lib/src/cordova/create.js [hunk 1]
```diff
@@ -220,8 +220,22 @@ function create(dir, optionalId, optionalName, cfg) {
...
         }
     })
-    .then(function(import_from_path) {
-
+    .then(function(input_directory) {
+        var import_from_path = input_directory;
+        var templatePkg;
+        
+        //handle package.json present with dirname
+        if (fs.existsSync(path.join(input_directory,'package.json'))){
```
Comment (vladimir-kotikov) [inline cordova-lib/src/cordova/create.js]: Can we omit this check? If there is no `package.json` in template directory we'll catch this in `catch (e)` below.

File: cordova-lib/src/cordova/create.js [hunk 2]
```diff
@@ -220,8 +220,22 @@ function create(dir, optionalId, optionalName, cfg) {
             }
         }
     })
-    .then(function(import_from_path) {
-
+    .then(function(input_directory) {
+        var import_from_path = input_directory;
+        var templatePkg;
```
Comment (vladimir-kotikov) [inline cordova-lib/src/cordova/create.js]: `templatePkg` is only used within `try` block and doesn't need to be defined in outer scope

**Comment by vladimir-kotikov** [comment]
LGTM apart from two minor nitpicks

**Comment by csantanapr** [comment]
@vladimir-kotikov thanks for the review I will check on those

**Comment by stevengill** [comment]
I don't think you should be running npm install carlos. This code doesn't even check if npm exists before trying to run it. 

I recommend you leaving the npm install to the fetch refactor (https://github.com/cordova/cordova-discuss/pull/33).

I'm not sure about this PR. I know you merged it in already. I think it requires more discussion.

**Comment by csantanapr** [comment]
@stevengill oops good catch, that npm install portion was not suppose to be there, I was just experimenting, only the top portion for the sub dir.

I agree npm install should be part of [cordova/cordova-discuss#33](https://github.com/cordova/cordova-discuss/pull/33) where everything is npm fetch and handles npm install

**Comment by csantanapr** [comment]
@stevengill the npm install portion is reverted now, I will take a look at your PR for npm fetch on how is handling the template fetching and npm install.

**Comment by stevengill** [comment]
still working on integrating it into the template work, but will add it
soon to the pr.

**Pull request closed without merging**